### PR TITLE
Issue 14711: Jakarta transformation for audit feature

### PR DESCRIPTION
**Pull request by kristip17**
Do Jarkarta updates for the audit-1.0 feature (and related internal features). Merge contents of private auditWebProbes into other existing audit features.

Enable Audit tests for JakartaEE.

### CODE REVIEW

**Comment by kristip17** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/protected/io.openliberty.auditCollector.1.0.feature [hunk 1]
```diff
@@ -0,0 +1,15 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/protected/io.openliberty.auditCollector.1.0.feature:1]: We don't need this new feature. The reason is, you can toleration servlet-5.0 in the existing com.ibm.websphere.appserver.auditCollector-1.0 feature.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.javaee.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+visibility = private
+
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.websphere.appserver.audit-1.0))",\
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.servlet-3.0)(osgi.identity=com.ibm.websphere.appserver.servlet-3.1)(osgi.identity=com.ibm.websphere.appserver.servlet-4.0)))"
+
+IBM-Install-Policy: when-satisfied
+
+-features=\
+  com.ibm.websphere.appserver.appSecurity-2.0; ibm.tolerates:="3.0", \
+  com.ibm.websphere.appserver.auditCollector-1.0
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.javaee.feature:13]: Remove the auditCollector-1.0 feature. We will leave it in the public feature.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.jakarta.feature [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+visibility = private
+
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.websphere.appserver.audit-1.0))",\
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.websphere.appserver.servlet-5.0))"
+
+IBM-Install-Policy: when-satisfied
+
+-features=\
+  io.openliberty.appSecurity-5.0, \
+  io.openliberty.auditCollector-1.0
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.jakarta.feature:13]: Remove the auditCollector-1.0 feature. We will leave it in the public feature.

File: dev/com.ibm.websphere.appserver.features/visibility/public/audit-1.0/com.ibm.websphere.appserver.audit-1.0.feature [hunk 4]
```diff
@@ -3,9 +3,6 @@ symbolicName=com.ibm.websphere.appserver.audit-1.0
 visibility=public
 IBM-ShortName: audit-1.0
 Subsystem-Name: Audit 1.0
--features=com.ibm.websphere.appserver.auditCollector-1.0, \
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/public/audit-1.0/com.ibm.websphere.appserver.audit-1.0.feature:6]: Leave the auditCollector-1.0 feature.

File: dev/com.ibm.websphere.appserver.features/visibility/public/audit-1.0/com.ibm.websphere.appserver.audit-1.0.feature [hunk 5]
```diff
@@ -3,9 +3,6 @@ symbolicName=com.ibm.websphere.appserver.audit-1.0
...
-com.ibm.websphere.appserver.appSecurity-2.0, \
-com.ibm.websphere.appserver.servlet-3.0; ibm.tolerates:="3.1,4.0"
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/public/audit-1.0/com.ibm.websphere.appserver.audit-1.0.feature:8]: Tolerate servlet-5.0.

com.ibm.websphere.appserver.servlet-3.0; ibm.tolerates:="3.1,4.0,5.0"

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.jakarta.feature (hunk 3 above)
Comment (jvanhill) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.jakarta.feature:13]: It is also possible that you may need to put a dependency on the servlet-5.0 feature.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.javaee.feature (hunk 2 above)
Comment (jvanhill) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.javaee.feature:13]: It is also possible that you may need to put a dependency on the servlet-3.0 feature (tolerating 3.1 and 4.0).

**Review by jvanhill — CHANGES_REQUESTED** [review]

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.jakarta.feature [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+kind=noship
+edition=core
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.jakarta.feature:16]: noship features need to be in the full edition.

File: dev/com.ibm.websphere.appserver.features/visibility/public/audit-1.0/com.ibm.websphere.appserver.audit-1.0.feature (hunk 4 above)
Comment (kristip17) [reply dev/com.ibm.websphere.appserver.features/visibility/public/audit-1.0/com.ibm.websphere.appserver.audit-1.0.feature:6]: Done

File: dev/com.ibm.websphere.appserver.features/visibility/public/audit-1.0/com.ibm.websphere.appserver.audit-1.0.feature (hunk 5 above)
Comment (kristip17) [reply dev/com.ibm.websphere.appserver.features/visibility/public/audit-1.0/com.ibm.websphere.appserver.audit-1.0.feature:8]: Done

**Comment by kristip17** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.jakarta.feature [hunk 7]
```diff
@@ -0,0 +1,18 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.audit-1.0.jakarta
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.jakarta.feature:3]: Seems convention is to remove the '-' in auto features. Applies to the name as well.

```suggestion
symbolicName = io.openliberty.audit1.0.jakarta
```

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.javaee.feature [hunk 8]
```diff
@@ -0,0 +1,17 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.audit-1.0.javaee
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit-1.0.javaee.feature:3]: Seems convention is to remove the '-' in auto features. Applies to the name as well.
```suggestion
symbolicName = io.openliberty.audit1.0.javaee
```

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditCollector-1.0.javaee.feature [hunk 9]
```diff
@@ -0,0 +1,15 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.auditCollector-1.0.javaee
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditCollector-1.0.javaee.feature:3]: Seems convention is to remove the '-' in auto features. Applies to the name as well.
```suggestion
symbolicName = io.openliberty.auditCollector1.0.javaee
```

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditCollector-1.0.jakarta.feature [hunk 10]
```diff
@@ -0,0 +1,15 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.auditCollector-1.0.jakarta
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditCollector-1.0.jakarta.feature:3]: Seems convention is to remove the '-' in auto features. Applies to the name as well.
```suggestion
symbolicName = io.openliberty.auditCollector1.0.jakarta
```

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditWebProbes-1.0.jakarta.feature [hunk 11]
```diff
@@ -0,0 +1,13 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName =io.openliberty.auditWebProbes-1.0.jakarta
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditWebProbes-1.0.jakarta.feature:3]: Seems convention is to remove the '-' in auto features. Applies to the name as well.
```suggestion
symbolicName =io.openliberty.auditWebProbes1.0.jakarta
```

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditWebProbes-1.0.javaee.feature [hunk 12]
```diff
@@ -0,0 +1,13 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName =io.openliberty.auditWebProbes-1.0.javaee
```
Comment (jvanhill) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditWebProbes-1.0.javaee.feature:3]: Seems convention is to remove the '-' in auto features. Applies to the name as well.
```suggestion
symbolicName =io.openliberty.auditWebProbes1.0.javaee
```

File: dev/com.ibm.ws.webcontainer.security.jacc.1.5_fat/publish/servers/com.ibm.ws.webcontainer.security.fat.basicauth.audit/server.xml [hunk 13]
```diff
@@ -12,7 +12,6 @@
 
 	<featureManager>
 		<feature>jsp-2.2</feature>
-		<feature>servlet-3.1</feature>
```
Comment (jvanhill) [inline dev/com.ibm.ws.webcontainer.security.jacc.1.5_fat/publish/servers/com.ibm.ws.webcontainer.security.fat.basicauth.audit/server.xml:15]: Not sure why this was removed. Can you explain?

**Review by jvanhill — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.webcontainer.security.jacc.1.5_fat/publish/servers/com.ibm.ws.webcontainer.security.fat.basicauth.audit/server.xml (hunk 13 above)
Comment (kristip17) [reply dev/com.ibm.ws.webcontainer.security.jacc.1.5_fat/publish/servers/com.ibm.ws.webcontainer.security.fat.basicauth.audit/server.xml]: I think I took it out earlier, put it back, test still passed.

**Comment by kristip17** [comment]
#build

**Review by jvanhill — APPROVED** [review]

**Comment by kristip17** [comment]
#libby

**Comment by kristip17** [comment]
Recent build had a failure on the auditWebProbes -- I removed the last filter, but didn't remove the last `\,` . Modify and rerunning.

**Comment by kristip17** [comment]
Latest run is  combo of this PR and a CL PR to resolve a feature checking issue -- will have to wait for CL PR to merge to master and rerun the build.

**Comment by kristip17** [comment]
#libby

**Comment by kristip17** [comment]
@jhanders34  -- Can you review the feature files please, when you are able?

**Comment by kristip17** [comment]
#build

**Comment by kristip17** [comment]
My build failed on the z/OS issue. Rerunning with `spawn.zos=false`

**Comment by kristip17** [comment]
Failures in PB look like known issues. Audit FATs are running -- including Jakarta versions.

Ready for final review.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit1.0.javaee.feature [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+symbolicName = io.openliberty.audit1.0.javaee
+visibility = private
+
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.websphere.appserver.audit-1.0))",\
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.servlet-3.0)(osgi.identity=com.ibm.websphere.appserver.servlet-3.1)(osgi.identity=com.ibm.websphere.appserver.servlet-4.0)))"
+
+IBM-Install-Policy: when-satisfied
+
+-features=\
+  com.ibm.websphere.appserver.appSecurity-2.0; ibm.tolerates:="3.0"
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit1.0.javaee.feature:12]: The tolerates 3.0 is new compared to the old config.  Was that intended?  Was it a bug before?

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditWebProbes1.0.jakarta.feature [hunk 15]
```diff
@@ -0,0 +1,13 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName =io.openliberty.auditWebProbes1.0.jakarta
+visibility = private
+IBM-Install-Policy: when-satisfied
+
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.audit1.0.jakarta))"
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditWebProbes1.0.jakarta.feature:7]: This filter doesn't make any sense.  Just move the bundles from this auto feature to the audit1.0.jakarta auto feature.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditWebProbes1.0.javaee.feature [hunk 16]
```diff
@@ -0,0 +1,13 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName =io.openliberty.auditWebProbes1.0.javaee
+visibility = private
+IBM-Install-Policy: when-satisfied
+
+IBM-Provision-Capability:  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.audit1.0.javaee))"
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.auditWebProbes1.0.javaee.feature:7]: This filter doesn't make any sense.  Just move the bundles from this auto feature to the audit1.0.javaee auto feature.

File: dev/com.ibm.websphere.appserver.features/visibility/protected/com.ibm.websphere.appserver.auditWebProbes-1.0.feature [hunk 17]
```diff
@@ -2,12 +2,10 @@
 symbolicName=com.ibm.websphere.appserver.auditWebProbes-1.0
 visibility=private
 IBM-Provision-Capability: \
-  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.servlet-3.0)(osgi.identity=com.ibm.websphere.appserver.servlet-3.1)(osgi.identity=com.ibm.websphere.appserver.servlet-4.0)))", \
-  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.appSecurity-1.0)(osgi.identity=com.ibm.websphere.appserver.appSecurity-2.0)(osgi.identity=com.ibm.websphere.appserver.appSecurity-3.0)))", \
-  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.websphere.appserver.audit-1.0))"
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.servlet-3.0)(osgi.identity=com.ibm.websphere.appserver.servlet-3.1)(osgi.identity=com.ibm.websphere.appserver.servlet-4.0)(osgi.identity=com.ibm.websphere.appserver.servlet-5.0)))", \
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/protected/com.ibm.websphere.appserver.auditWebProbes-1.0.feature:5]: This protected feature which is really an auto feature doesn't make any sense.  audit already depends on servlet and appSecurity.  As such this file should be deleted and the requestProbes dependency should just be added to audit-1.0 feature.

File: dev/com.ibm.ws.security.audit.reader.auditreader/bnd.bnd [hunk 18]
```diff
@@ -13,6 +13,8 @@ bVersion=1.0
 Bundle-Name: WebSphere Audit Reader Utility
 Bundle-SymbolicName: com.ibm.ws.security.audit.reader.auditreader
 
+jakartaeeMe: true
```
Comment (jhanders34) [inline dev/com.ibm.ws.security.audit.reader.auditreader/bnd.bnd:16]: auditreader doesn't appear to have any Java EE dependencies unless I am missing something.  Why does it need to be jakarta-ized?

**Review by jhanders34 — CHANGES_REQUESTED** [review]

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit1.0.javaee.feature (hunk 14 above)
Comment (kristip17) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.audit1.0.javaee.feature:12]: Yes, this was a bug before.

**Review by jhanders34 — APPROVED** [review]

**Comment by kristip17** [comment]
#build

**Comment by kristip17** [comment]
#build

**Comment by kristip17** [comment]
#libby

**Pull request merged**